addons/point_of_sale/models/pos_order.py

```diff
@@ -483,6 +483,7 @@ def create(self, vals_list):
     def _complete_values_from_session(self, session, values):
         if values.get('state') and values['state'] == 'paid' and not values.get('name'):
             values['name'] = self._compute_order_name()
+        values.pop('pricelist_id', False)   # pricelist_id is False in self-ordering and kiosk modes
```

Comment:
For self-ordering and kiosk mode, when the pricelist_id is false, the setdefault function doesn't change the pricelist_id because it defaults to false. Therefore, we need to remove the pricelist_id and have setdefault automatically set the pricelist_id.

Reply:
The problem is that, this is working fine in a kiosk or self scenario but this line is breaking the classic shop or restaurant scenario. You can test it, in a shop you make multiple pricelist available, then use one that is thge non-default one for an order, and then go to the backend on the order to look at its pricelist. It won't be the one you chosed. A better solution would be to set the pricelist for the order in the `PosSelfOrderController` in the `process_new_order ` method. Add `'pricelist_id': pos_config.pricelist_id.id if pos_config.pricelist_id else False` to the order dict in the method and that should solve your problem.